examples/light-switch-app/efr32/src/main.cpp

```diff
@@ -0,0 +1,220 @@
+#include <openthread/heap.h>
+#include <openthread/icmp6.h>
+#include <openthread/instance.h>
+#include <openthread/link.h>
+#include <openthread/platform/openthread-system.h>
+#include <openthread/tasklet.h>
+#include <openthread/thread.h>
+#endif // CHIP_ENABLE_OPENTHREAD
+
+#if defined(RS911X_WIFI) || defined(WF200_WIFI)
+#include "wfx_host_events.h"
+#endif /* RS911X_WIFI */
+
+#if PW_RPC_ENABLED
+#include "Rpc.h"
+#endif
+
+#ifdef ENABLE_CHIP_SHELL
+#include "matter_shell.h"
+#endif
+
+#define BLE_DEV_NAME "SiLabs-Light-Switch"
+using namespace ::chip;
+using namespace ::chip::Inet;
+using namespace ::chip::DeviceLayer;
+
+#define UNUSED_PARAMETER(a) (a = a)
+
+volatile int apperror_cnt;
+// ================================================================================
+// App Error
+//=================================================================================
+void appError(int err)
+{
+    EFR32_LOG("!!!!!!!!!!!! App Critical Error: %d !!!!!!!!!!!", err);
+    portDISABLE_INTERRUPTS();
+    while (1)
+        ;
+}
+
+void appError(CHIP_ERROR error)
+{
+    appError(static_cast<int>(error.AsInteger()));
+}
+
+// ================================================================================
+// FreeRTOS Callbacks
+// ================================================================================
+extern "C" void vApplicationIdleHook(void)
+{
+    // FreeRTOS Idle callback
+
+    // Check CHIP Config nvm3 and repack flash if necessary.
+    Internal::EFR32Config::RepackNvm3Flash();
+}
+
+// ================================================================================
+// Main Code
+// ================================================================================
+int main(void)
+{
+    init_efrPlatform();
+    mbedtls_platform_set_calloc_free(CHIPPlatformMemoryCalloc, CHIPPlatformMemoryFree);
+
+#if PW_RPC_ENABLED
+    chip::rpc::Init();
+#endif
+
+#ifdef HEAP_MONITORING
+    MemMonitoring::startHeapMonitoring();
+#endif
+
+    EFR32_LOG("==================================================");
+    EFR32_LOG("chip-efr32-light-switch-example starting");
+    EFR32_LOG("==================================================");
+
+    EFR32_LOG("Init CHIP Stack");
+    // Init Chip memory management before the stack
+    chip::Platform::MemoryInit();
+    chip::DeviceLayer::PersistedStorage::KeyValueStoreMgrImpl().Init();
+
+    CHIP_ERROR ret = PlatformMgr().InitChipStack();
+    if (ret != CHIP_NO_ERROR)
+    {
+        EFR32_LOG("PlatformMgr().InitChipStack() failed");
+        appError(ret);
+    }
+    chip::DeviceLayer::ConnectivityMgr().SetBLEDeviceName(BLE_DEV_NAME);
+#if CHIP_ENABLE_OPENTHREAD
+    EFR32_LOG("Initializing OpenThread stack");
+    ret = ThreadStackMgr().InitThreadStack();
+    if (ret != CHIP_NO_ERROR)
+    {
+        EFR32_LOG("ThreadStackMgr().InitThreadStack() failed");
+        appError(ret);
+    }
+
+    ret = ConnectivityMgr().SetThreadDeviceType(ConnectivityManager::kThreadDeviceType_Router);
```

Comment:
FTD/MTD/SED type selection to be done here

Reply:
done